pkg/controllermanager/controller/shoot/shoot_maintenance_control.go

```diff
@@ -224,3 +228,70 @@ func hasMaintainNowAnnotation(shoot *gardenv1beta1.Shoot) bool {
+	machineImagesFound, machineImages, err := helper.DetermineMachineImages(*cloudProfile, shootCurrentImage.Name)
+	if err != nil {
+		return false, nil, fmt.Errorf("failure while determining the machine images in the CloudProfile: %s", err.Error())
+	}
+	if !machineImagesFound || len(machineImages) == 0 {
+		return false, nil, fmt.Errorf("failure while determining the machine images in the CloudProfile: no machineImage with name '%s' (specified in shoot) could be found in the cloud cloudProfile '%s'", shootCurrentImage.Name, cloudProfile.Name)
+	}
+
+	// checks if force update is required & applicable (has the specified version)
+	forceUpdateRequired := ForceUpdateRequired(shootCurrentImage, machineImages)
+
+	// safely dereference due to defaulting in api server
+	autoUpdateOfMachineImageConfigured := *shoot.Spec.Maintenance.AutoUpdate.MachineImageVersion
+	if forceUpdateRequired || autoUpdateOfMachineImageConfigured {
+		versionUpdatePossible, updateMachineImage, err := VersionUpdatePossible(shootCurrentImage, machineImages)
+		if err != nil {
+			return false, nil, err
+		}
+		if versionUpdatePossible {
+			return true, updateMachineImage, nil
+		}
+	}
+	return false, nil, nil
+}
+
+// VersionUpdatePossible checks if the shoot's current image can be updated to a higher semantic version
+func VersionUpdatePossible(shootCurrentImage *gardenv1beta1.MachineImage, machineImages []gardenv1beta1.MachineImage) (bool, *gardenv1beta1.MachineImage, error) {
+	latestSemVerVersion, latestMachineImage, err := helper.DetermineLatestMachineImage(machineImages)
+	if err != nil {
+		return false, nil, fmt.Errorf("failed to determine latest machine image in cloud profile: %s", err.Error())
+	}
+
+	shootImageSemVerVersion, err := semver.NewVersion(shootCurrentImage.Version)
+	if err != nil {
+		return false, nil, fmt.Errorf("error while parsing machine image version of shoot (invalid): %s", err.Error())
+	}
+
+	if latestSemVerVersion.GreaterThan(shootImageSemVerVersion) {
+		return true, &latestMachineImage, nil
+	}
+	return false, nil, nil
+}
+
+// ForceUpdateRequired checks if the shoots current machine image has to be forcefully updated
+func ForceUpdateRequired(shootCurrentImage *gardenv1beta1.MachineImage, imagesCloudProvider []gardenv1beta1.MachineImage) bool {
+	// machine images are already from the same name
+	for _, image := range imagesCloudProvider {
+		if image.ExpirationDate == nil {
+			continue
+		}
+
+		updateTimeApplies := ExpirationDateApplies(*image.ExpirationDate)
+
+		if shootCurrentImage.Version == image.Version && updateTimeApplies {
+			return true
+		}
+	}
+	return false
+}
+
+// ExpirationDateApplies returns if now is equal or after the given expirationDate
+func ExpirationDateApplies(timestamp metav1.Time) bool {
+	return time.Now().After(timestamp.Time) || time.Now().Equal(timestamp.Time)
```

Comment:
You should make sure to convert everything (now, and the read time) to UTC before comparing anything.